Added Physical Robot Setup Docs + Example for Using Physical Robot without Remote Controller (#359)
ci-group/revolve2: docs/source/physical_robot_installation_guide/index.rst

> ===================================

Reviewer: Somewhere early on you need to mention that Raspberry Pi is abbreviated as RPi. I know you had that before but it must have accidentally disappeared

--------------------
Install the Hardware
--------------------
For any type of hardware there are some required items:

* A controller such as a RPi.
* A HAT for the RPi to connect servos and other extensions to.
* Modules, Servos and extensions that have to be mounted.
* Cables and a Battery

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Hardware setup for a V1 Robot
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
**This is only applicable to the CI-Group specific V1 design.**
Print the Modules and install the RPi in the Head compartment.
Assemble the robot and connect the Servos to the HAT.
More instructions can be found in the *how-to-build-a-modular-robot-guide*.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Hardware setup for a V2 Robot
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
**This is only applicable to the CI-Group specific V2 design.**
To build a V2 robot print all the necessary Modules and mount the RPi in the Head Compartment.
For instructions about the HAT and the Robot architecture itself refer to `<https://github.com/ci-group/robohat>`_.
Assemble the robot and connect the Servos to the HAT.

------------------
Setting up the RPi
------------------
This step is the same for all types of hardware.

> To setup the RPI follow the steps below carefully.

Reviewer: ```suggestion
To setup the RPi follow the steps below carefully.
```

#. Flash the SD-card with the RPi OS (previously Raspbian). Some Important notes:
    * Reminder for CI Group people: if you select a keyboard layout, pick `English(US)`. It can always be changed using the `raspi-config` tool on the RPi.

>     * When setting up the WIFI connection, select your Country. (CI-Group WIFI: *ThymioNet*)

Reviewer: ![image](https://github.com/ci-group/revolve2/assets/10170564/82339ca1-defb-4070-a1a0-e5d7c010eaaa)
weird word, I never remember

> #. Boot the RPi and get CLI-access. Here are some general Tips:

Reviewer: ```suggestion
#. Boot the RPi and get command line access. Here are some general Tips:
```
We only use this word once anyway

    * You can connect a keyboard and screen.

>     * Better is to SSH. This work only if you and the RPi are on the same network. For CI Group, connect it to *ThymioNet* Wi-Fi.

Reviewer: ```suggestion
    * Better is to SSH. This works only if you and the RPi are on the same network. For CI Group, connect it to *ThymioNet* Wi-Fi.
```

>     * If you want to SSH and dont know the IP of the RPi use: :code:`sudo nmap -sP <your ip>` to find all clients on your network.

Reviewer: ```suggestion
    * If you want to SSH and dont know the IP of the RPi, you can use :code:`sudo nmap -sP <your ip>` to find all clients on your network.
```

---------------------------
Install Revolve2 on the RPi
---------------------------

> Setting up Revolve2 on the robot requires different steps, depending on the hardware version. Some general steps for all versions:

Reviewer: This first sentence appears bold in my rendering and I would like it to not be bold. But I don't see why it renders as bold? You didn't put any stars around it. weird

Author: i know, i tried to get rid of it but  didnt find a way yet

>     * Setup a global pyenv. (This is necessary because RPis dont allow global pip installs):

Reviewer: can we do
1.
2.
like you did in the previous step

>     * Setup a global pyenv. (This is necessary because RPis dont allow global pip installs):

Reviewer: ```suggestion
    * Setup a global pyenv. This is to prevent changes to the system's Python installation.:
```

>         #. Install pyenv: :code:`curl https://pyenv.run | bash` & :code:`git clone https://github.com/pyenv/pyenv-virtualenv.git $(pyenv root)/plugins/pyenv-virtualenv`

Reviewer: so we need to check on a new RPi if the virtualenv installation is needed or if it comes with pyenv by default.

        #. Add :code:`pyenv` to bash:
            .. code-block:: bash

                echo '# Enable pyenv
                export PYENV_ROOT="$HOME/.pyenv"
                command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"
                eval "$(pyenv init -)"
                eval "$(pyenv virtualenv-init -)"
                ' >> ~/.bashrc
        #. Log in and out of the rpi.

>         #. Get the right python version (this takes a long time): :code:`pyenv install 3.11`

Reviewer: sudo apt install -y \
    libssl-dev \
    libbz2-dev \
    libncurses5-dev \
    libncursesw5-dev \
    libreadline-dev \
    libsqlite3-dev \
    libffi-dev \
    liblzma-dev
pyenv install 3.11

the required packages are important

        #. create a global virtualenv: :code:`pyenv virtualenv 3.11 global_env` & :code:`pyenv global global_env`
    * Then install using :code:`pip install git+https://github.com/ci-group/revolve2.git`.

>         * If you use a CI-Group specific hardware such as V1/ V2 do the install as follows: :code:`pip install <dir>/revolve2/modular_robot_physical[<botv1 or botv2>]`

Reviewer: pip install "git+https://github.com/ci-group/revolve2.git@314-rewrite-rpi#egg=revolve2-modular_robot_physical[botv1]&subdirectory=modular_robot_physical"

Reviewer: all these directive are important sadly. once it's on pypi it will be easier

^^^^^^^^^^^^^^^^^^^
V1 Additional Steps
^^^^^^^^^^^^^^^^^^^
If you use V1 hardware setup requires additional steps:

> * Enabling :code:`pigpiod` deamon so it enables with startup (used for the servos).

Reviewer: you should start by saying you need to install pigpiod :) maybe with a one sentence explanation about what it is

    #. Add the following content to the file:
        .. code-block:: bash

            ini
            [Unit]
            Description=Pigpio Daemon
            After=network-online.target

            [Service]
            Type=forking
            ExecStart=/usr/bin/pigpiod -l -n localhost
            ExecStop=/bin/kill -15 $MAINPID
            Nice=-10
            Restart=on-failure
            RestartSec=10

            [Install]
            WantedBy=multi-user.target
    #. Here, the :code:`Nice=-10` line sets a high priority for the daemon (lower values are higher priority, with -20 being the highest priority). The :code:`-l` option in the :code:`ExecStart` line tells :code:`pigpiod` to only listen on the localhost interface. The :code:`-n localhost` option ensures that pigpiod only runs if it can connect to localhost (preventing certain failure cases).
    #. Enable and start the service: :code:`sudo systemctl daemon-reload` & :code:`sudo systemctl enable pigpiod` & :code:`sudo systemctl start pigpiod`.
    #. Check if it is running properly using: :code:`sudo systemctl status pigpiod`

> That's it! Now :code:`pigpiod` will run at startup with a high priority. If you need to adjust the priority later, you can edit the :code:`Nice` value in the service file and restart the service.

Reviewer: this paragraph is not indented but it should be i think

===============================
Setting up a new Physical Robot
===============================
This tutorial is aimed towards developers, who aim to set up a new robot.
If you simply want to know how to run a physical robot, check the examples.

-----------
Preparation
-----------
Firstly make sure to have all materials necessary to build the physical robot.
Models to print the Modules can be found on the following git repo: `<https://github.com/ci-group/revolve-models>`_.
Other items needed:

* Rasberry Pi (rpi)
* SD-card for the rpi-OS
* suitable Battery
* HAT for the rpi
* cables

------------------
Setting up the RPI
------------------
To setup the RPI follow the steps below carefully.

#. Flash the SD-card with the RPI-OS. Some Important notes:
    * Make sure to select a appropriate Keyboard Layout (preferably English(US)).

    * When setting up the WIFI connection, select your Country.

> #. Once the SD-card is flashed, put it into the rpi, mounted on the robot and connect to it. If you dont know the IP-address do the following:

Reviewer: ```suggestion
#. Boot the RPi and get command line access. Here are some general tips.
    * You can connect a keyboard and screen.
    * Better is to SSH. This work only if you and the RPi are on the same network. For CI Group, connect it to ThymioNet Wi-Fi.
    * You can find the IP addresses of devices on your network through :code:`sudo nmap -sP <your ip>`.
```

I think we can expect a certain level of expertise from people setting up these brains. i would hope they know `ip` haha

    * :code:`ip a` to find your ip in the **same** network.

    * :code:`sudo nmap -sP <your ip>` to find all clients on your network.

    * Find your rpi and connect via ssh.

#. Setting up Revolve2 on the robot requires different steps, depending on the hardware version. Some general steps for all versions:

>     * In order to install Revolve2 you either have to create a venv or use a global pyenv, since rpi doesnt allow to pip install packages globally.

Reviewer: We do not want this to be a choice. We pick pyenv to prevent confusion

        * Make and activate a venv on the rpi: :code:`python -m venv .venv` & :code:`source .venv/bin/activate`

        * For :code:`pyenv` follow the steps below:
            #. Install pyenv: :code:`curl https://pyenv.run | bash` & :code:`git clone https://github.com/pyenv/pyenv-virtualenv.git $(pyenv root)/plugins/pyenv-virtualenv`

            #. Add :code:`pyenv` to bash:
                .. code-block:: bash

                    echo '# Enable pyenv
                    export PYENV_ROOT="$HOME/.pyenv"
                    command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"
                    eval "$(pyenv init -)"
                    eval "$(pyenv virtualenv-init -)"
                    ' >> ~/.bashrc

            #. Log in and out of the rpi.

            #. Get the right python version (this takes a long time): :code:`pyenv install 3.11`

            #. create a global virtualenv: :code:`pyenv virtualenv 3.11 global_env` & :code:`pyenv global global_env`

>     * Clone the Revolve2 branch or repository for your installation.

Reviewer: We do not clone and directly pip install from github instead to prevent mistakes

Reviewer: once it's on pypi it will be even shorter
